mpisppy/tests/test_ci.py

```diff
@@ -0,0 +1,222 @@
+# Copyright 2020 by B. Knueven, D. Mildebrath, C. Muir, J-P Watson, and D.L. Woodruff
+# This software is distributed under the 3-clause BSD License.
+# Provide some test for confidence intervals
+"""
+
+"""
+
+import os
+import tempfile
+import numpy as np
+import unittest
+
+from math import log10, floor
+import pyomo.environ as pyo
+import mpi4py.MPI as mpi
+
+
+import mpisppy.tests.examples.farmer as farmer 
+
+
+import mpisppy.utils.MMWci as MMWci
+import mpisppy.utils.amalgomator as ama
+from mpisppy.utils.xhat_eval import Xhat_Eval
+
+fullcomm = mpi.COMM_WORLD
+global_rank = fullcomm.Get_rank()
+
+__version__ = 0.02
+
+solvers = [n+e for e in ('_persistent', '') for n in ("cplex","gurobi","xpress")]
+
+
+for solvername in solvers:
+    solver_available = pyo.SolverFactory(solvername).available()
+    if solver_available:
+        break
+
+
+if '_persistent' in solvername:
+    persistentsolvername = solvername
+else:
+    persistentsolvername = solvername+"_persistent"
+try:
+    persistent_available = pyo.SolverFactory(persistentsolvername).available()
+except:
+    persistent_available = False
```

Comment:
For maintainability I would prefer we import this logic from test_ef_ph.py or import both from a common source as opposed to copy-pasting it in.

Reply:
Done